modules/core/src/main/java/org/apache/ignite/internal/binary/compression/compressors/CompressionUtils.java

```diff
@@ -0,0 +1,120 @@
+     * @throws BinaryObjectException if failed to compress bytes.
+     */
+    public static byte[] compress(BinaryContext context, CompressionType type,
+        byte[] bytes) throws BinaryObjectException {
+        Map<CompressionType, Compressor> compressorsSelector = context.configuration().getCompressorsSelector();
+
+        return compress(compressorsSelector.get(type), bytes);
+    }
+
+    /**
+     * Compresses given bytes, using given {@link Compressor}.
+     *
+     * @param compressor Compressor for compressing.
+     * @param bytes Non compressed bytes.
+     * @return Compressed bytes.
+     * @throws BinaryObjectException if failed to compress bytes.
+     */
+    public static byte[] compress(Compressor compressor, byte[] bytes) throws BinaryObjectException {
+        try {
+            return compressor.compress(bytes);
+        }
+        catch (IOException e) {
+            throw new BinaryObjectException("Failed to compress bytes", e);
+        }
+    }
+
+    /**
+     * Decompresses given bytes, using {@link Compressor} which defined in BinaryContext
+     * and mapped with given {@link CompressionType}.
+     *
+     * @param context BinaryContext.
+     * @param type CompressionType.
+     * @param bytes Compressed bytes.
+     * @return Decompressed bytes.
+     * @throws BinaryObjectException if failed to decompress bytes.
+     */
+    public static byte[] decompress(BinaryContext context, CompressionType type,
+        byte[] bytes) throws BinaryObjectException {
+        Map<CompressionType, Compressor> compressorsSelector = context.configuration().getCompressorsSelector();
+
+        return decompress(compressorsSelector.get(type), bytes);
+    }
+
+    /**
+     * Decompresses given bytes, using given {@link Compressor}.
+     *
+     * @param compressor Compressor for decompressing.
+     * @param bytes Compressed bytes.
+     * @return Decompressed bytes.
+     * @throws BinaryObjectException if failed to decompress bytes.
+     */
+    public static byte[] decompress(Compressor compressor, byte[] bytes) throws BinaryObjectException {
+        try {
+            return compressor.decompress(bytes);
+        }
+        catch (IOException e) {
+            throw new BinaryObjectException("Failed to decompress bytes", e);
+        }
+    }
+
+    /**
+     * Check if type is compressed.
+     *
+     * @param mode BinaryWriteMode.
+     * @return 'true' if type is compressed, otherwise 'false'.
+     */
+    public static boolean isCompressionType(BinaryWriteMode mode) {
+        return isCompressionType(mode.typeId());
+    }
+
+    /**
+     * Check if type is compressed.
+     *
+     * @param typeId Type id defined in the {@link GridBinaryMarshaller}.
+     * @return 'true' if type is compressed, otherwise 'false'.
+     */
+    public static boolean isCompressionType(int typeId) {
+        return typeId >= GridBinaryMarshaller.GZIPPED && typeId <= GridBinaryMarshaller.COMPRESSED_CUSTOM;
```

Comment:
what if someone else add certain uncompressed type between GZIPPED and COMPRESSED_CUSTOM ?